symphony-bdk-core/src/main/java/com/symphony/bdk/core/client/BearerEnabledApiClient.java

```diff
@@ -0,0 +1,123 @@
+package com.symphony.bdk.core.client;
+
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiResponse;
+import com.symphony.bdk.http.api.Pair;
+import com.symphony.bdk.http.api.auth.Authentication;
+import com.symphony.bdk.http.api.auth.OAuth;
+import com.symphony.bdk.http.api.util.TypeReference;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An {@link ApiClient} implementation used when common jwt feature is enabled.
+ * It contains an {@link ApiClient} that will allow the usage of the Authentication header in each API call.
+ * It will also remove the sessionToken header from the request.
+ * Before adding the jwt as Authorization header the client is checking the jwt expiration, if the jwt is expired
+ * re-authentication will be triggered and the jwt will be refreshed.
+ */
+
+@Slf4j
+@API(status = API.Status.EXPERIMENTAL)
+public class BearerEnabledApiClient implements ApiClient {
```

Comment:
I think we could make it a bit more generic from an `ApiClient` API perspective. I'd prefer something like: 
```java
ApiClient apiClient = apiClientFactory.getPodClient();
if (isCommonJwtEnabled()) {
  apiClient.getAuthentications().put("bearerAuth", new OAuth(this::retrieveBearerAuthToken))
  apiClient.enforceAuthenticationScheme("bearerAuth");
}
```
This way we would have to modify a bit the logic [here](https://github.com/finos/symphony-bdk-java/blob/main/symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/ApiClientJersey2.java#L555), to apply `Authentication` impl based on an `enforcedAuthenticationScheme` class attribute.

Reply:
done here https://github.com/finos/symphony-bdk-java/pull/618/commits/9c5d6a37ce348f39cc9a8a5da5f738412250dae3